x-pack/plugins/cloud_defend/server/lib/telemetry/collectors/installation_stats_collector.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import yaml from 'js-yaml';
+import type { CoreStart, Logger, SavedObjectsClientContract } from '@kbn/core/server';
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import {
+  AgentPolicy,
+  PackagePolicy,
+  PACKAGE_POLICY_SAVED_OBJECT_TYPE,
+  SO_SEARCH_LIMIT,
+} from '@kbn/fleet-plugin/common';
+import { agentPolicyService } from '@kbn/fleet-plugin/server/services';
+import type { CloudDefendInstallationStats } from './types';
+import type { CloudDefendPluginStart, CloudDefendPluginStartDeps } from '../../../types';
+import { INTEGRATION_PACKAGE_NAME, INPUT_CONTROL } from '../../../../common/constants';
+import { getInputFromPolicy } from '../../../../common/utils/helpers';
+
+export const getInstallationStats = async (
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract,
+  coreServices: Promise<[CoreStart, CloudDefendPluginStartDeps, CloudDefendPluginStart]>,
+  logger: Logger
+): Promise<CloudDefendInstallationStats[]> => {
+  const [, cloudDefendServerPluginStartDeps] = await coreServices;
+
+  const cloudDefendContext = {
+    logger,
+    esClient,
+    soClient,
+    agentPolicyService: cloudDefendServerPluginStartDeps.fleet.agentPolicyService,
+    packagePolicyService: cloudDefendServerPluginStartDeps.fleet.packagePolicyService,
+  };
+
+  const getInstalledPackagePolicies = async (
+    packagePolicies: PackagePolicy[],
+    agentPolicies: AgentPolicy[]
+  ) => {
+    const installationStats = packagePolicies.map(
+      (packagePolicy: PackagePolicy): CloudDefendInstallationStats => {
+        const agentCounts =
+          agentPolicies?.find((agentPolicy) => agentPolicy?.id === packagePolicy.policy_id)
+            ?.agents ?? 0;
+
+        const input = getInputFromPolicy(packagePolicy, INPUT_CONTROL);
+        const policyYaml = input?.vars?.configuration?.value;
+        let policyJson: any = {};
+
+        try {
+          policyJson = JSON.stringify(yaml.load(policyYaml));
+        } catch (err) {
+          logger.error(
+            'could not convert cloud_defend yaml to json for installation stats usage collection',
+            err
+          );
+        }
+
+        return {
+          package_policy_id: packagePolicy.id,
+          policy_yaml: policyYaml,
+          policy_json: policyJson,
+          package_version: packagePolicy.package?.version as string,
+          created_at: packagePolicy.created_at,
+          agent_policy_id: packagePolicy.policy_id,
+          agent_count: agentCounts,
+        };
```

Comment:
Is it possible to track the deployment environment for **dfc** integration installation stats?

Reply:
Do you mean 'aws' vs 'gcloud' vs 'azure'?

Reply:
cloud_provider has been added to accounts_stats